### DISCUSSION TITLE
Error during setup -- Pre-compiling FireMarshal buildroot sources

### DISCUSSION THREAD

**Discussion by FDG31** [ucb-bar/chipyard#2152] [Q&A]
Hi, I'm following the instruction in [https://chipyard.readthedocs.io/en/latest/Chipyard-Basics/Initial-Repo-Setup.html](url) and when I run the _./build-setup.sh riscv-tools_  it returns this:

ERROR: Failed to build workload br-base.json
Log available at: /home/fdg/chipyard/software/firemarshal/logs/br-base-build-2024-12-12--11-44-28-DCN23T9UH8ME7WRE.log
ERROR: FAILURE: 1 builds failed
build-setup.sh: Build script failed with exit code 1 at step 9: Pre-compiling FireMarshal buildroot sources.

This is the complete log file:
[br-base-build-2024-12-12--11-44-28-DCN23T9UH8ME7WRE.log](https://github.com/user-attachments/files/18111039/br-base-build-2024-12-12--11-44-28-DCN23T9UH8ME7WRE.log)

What do you suggest me to do?

**Comment by jerryz123** [comment]
There is a flag in that script which will skip firemarshal

**Reply by FDG31** [reply]
But firemarshal isn't a necessary part for work with the ChipYard framework?

**Reply by jerryz123** [reply]
It only is if you want to build Linux images

**Reply by FDG31** [reply]
Ok, so I should append the flag -s 8 -s 9 to the setup code and just wait the finishing. I thank you for this. 
Instead for the problem itself do you know what is the cause? I'm curious, maybe I have some files in my operating system missing or something that is creating this problem.
